featuretools/entityset/base_entity.py

```diff
@@ -124,6 +125,17 @@ def __eq__(self, other, deep=False):
             for v in self.variables:
                 if v not in other.variables:
                     return False
+            if self.indexed_by is None and other.indexed_by is not None:
+                return False
+            else:
+                for v, index_map in self.indexed_by.items():
+                    if v not in other.indexed_by:
+                        return False
+                    for i, related in index_map.items():
+                        if i not in other.indexed_by[v]:
+                            return False
+                        if not (np.sort(related) == np.sort(other.indexed_by[v][i])).all():
```

Comment:
can you leave a comment in the code what this line is checking?

Reply:
done